c++/h5/scalar.hpp

```diff
@@ -30,7 +31,18 @@ namespace h5 {
         return;
       }
     }
-    array_interface::read(g, name, array_interface::h5_array_view_from_scalar(x), array_interface::get_h5_lengths_type(g, name));
+
+    auto lt = array_interface::get_h5_lengths_type(g, name);
+
+    if constexpr (is_complex_v<T>) {
+      // Allow reading complex as a compound hdf5 dataype
+      if (hdf5_type_equal(lt.ty, hdf5_type<cplx_t>())) {
+        h5_read(g, name, reinterpret_cast<cplx_t &>(x));
```

Comment:
reinterpret_cast : most likely correct for complex, 
you can also simply copy to be very safe, the optimizer will remove the copy anyway